src/Twig/EventRuntime.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace AppBundle\Twig;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Event;
+use AppBundle\Repository\EventRegistrationRepository;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+class EventRuntime
+{
+    private $eventRegistrationRepository;
+
+    public function __construct(EventRegistrationRepository $eventRegistrationRepository)
+    {
+        $this->eventRegistrationRepository = $eventRegistrationRepository;
+    }
+
+    public function isEventAlreadyParticipating(Event $event, ?UserInterface $user): bool
+    {
+        if (!$user instanceof Adherent) {
+            return false;
+        }
+
+        return $this->eventRegistrationRepository->findGuestRegistration($event->getUuid(), $user->getEmailAddress()) ? true : false;
```

Comment:
as an alternative:
```
return (bool) $this->eventRegistrationRepository->findGuestRegistration(
    $event->getUuid(), 
    $user->getEmailAddress()
);
```

Reply:
I prefer don't cast the result to be sure to control the return.

Reply:
it is the exactly what you are doing with 
```
return $object ? true : false;
=== 
return (bool) object;
```
to avaluate the result of `if ($object)` PHP cast the $object in boolean value.